src/spectral/utils/spectral-validator.js

```diff
@@ -0,0 +1,44 @@
+const MessageCarrier = require('../../plugins/utils/messageCarrier');
+
+const parseResults = function(results) {
+  const messages = new MessageCarrier();
+
+  if (results) {
+    for (const validationResult of results) {
+      if (validationResult) {
+        const code = validationResult['code'];
+        const severity = validationResult['severity'];
+        const message = validationResult['message'];
+        const path = validationResult['path'];
+
+        if (code && severity != null && message && path) {
```

Comment:
@dpopp07 - my Node is rusty, but I hope simply ANDing each variable is the equivalent of ``!= null && != undefined``

Reply:
That is correct, although I would probably want to check more than just `null` for `severity`. Maybe something like `typeof severity === 'number'`